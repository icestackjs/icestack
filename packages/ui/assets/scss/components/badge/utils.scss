@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("badge.selector");
$DefaultMap: inject("badge.defaults.utils");

#{$S} {
  @each $typeName, $map in map.get($DefaultMap, "sizes") {
    &#{getSelector($typeName)} {
      @include resolve(map.get($map, "default"));
    }
  }
}
