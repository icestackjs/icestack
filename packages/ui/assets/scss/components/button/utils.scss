@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("button.selector");
$DefaultMap: inject("button.defaults.utils");

#{$S} {
  @each $size, $map in map.get($DefaultMap, "sizes") {
    &#{getSelector($size)} {
      @include resolve(map.get($map, "default"));
    }
  }

  @each $shape, $map in map.get($DefaultMap, "shapes") {
    &#{getSelector($shape)} {
      @include resolve(map.get($map, "default"));

      @each $size, $m in map.get($map, "sizes") {
        &:where(#{getSelector($size,#{$S}-)}) {
          @include resolve(map.get($m, "default"));
        }
      }
    }
  }
}
