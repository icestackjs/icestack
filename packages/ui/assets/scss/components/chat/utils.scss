@use "../../utils/index.scss" as *;

@include spread(inject("chat.defaults.utils"));
