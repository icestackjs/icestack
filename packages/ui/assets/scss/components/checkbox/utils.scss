@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("checkbox.selector");
$DefaultMap: inject("checkbox.defaults.utils");

@mixin utils {
  @each $s, $v in map.get($DefaultMap, "sizes") {
    &#{getSelector($s)} {
      @include resolve(map.get($v, "default"));
    }
  }
}

#{$S} {
  @include utils;
}
