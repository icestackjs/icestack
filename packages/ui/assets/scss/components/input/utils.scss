@use "../../utils/index.scss" as *;

@include spread(inject("input.defaults.utils"));
