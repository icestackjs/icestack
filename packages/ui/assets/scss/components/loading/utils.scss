@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("loading.selector");
$DefaultMap: inject("loading.defaults.utils");

#{$S} {
  @each $s, $v in map.get($DefaultMap, "sizes") {
    &#{getSelector($s)} {
      @include resolve($v);
    }
  }

  @each $s, $v in map.get($DefaultMap, "shapes") {
    &#{getSelector($s)} {
      @include resolve($v);
    }
  }
}
