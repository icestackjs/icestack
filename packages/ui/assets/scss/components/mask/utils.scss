@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("mask.selector");
$DefaultMap: inject("mask.defaults.utils");

#{$S} {
  @each $s, $v in map.get($DefaultMap, "shapes") {
    &#{getSelector($s)} {
      @include resolve($v);
    }
  }
}
