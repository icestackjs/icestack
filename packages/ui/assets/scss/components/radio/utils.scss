@use "../../utils/index.scss" as *;
@use "sass:map";

$D: inject("radio.defaults.utils");
$S: inject("radio.selector");

#{$S} {
  @each $s, $v in map.get($D, "sizes") {
    &#{getSelector($s)} {
      @include resolve(map.get($v, "default"));
    }
  }
}
