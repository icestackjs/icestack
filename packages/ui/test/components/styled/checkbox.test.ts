import { resolve } from './utils'
import { compileScss } from '@/sass'
describe('checkbox', () => {
  it('snap', async () => {
    const css = await compileScss(resolve('checkbox'))
    expect(css).toMatchSnapshot()
  })
})
