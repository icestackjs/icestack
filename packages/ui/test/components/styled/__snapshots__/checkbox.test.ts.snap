// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`checkbox > snap 1`] = `
".checkbox {
  @apply border-base-content rounded-btn h-6 w-6 cursor-pointer appearance-none border border-opacity-20;
}
.checkbox:focus-visible {
  @apply outline-base-content outline outline-2 outline-offset-2;
}
.checkbox:checked, .checkbox[checked=true], .checkbox[aria-checked=true] {
  @apply bg-base-content bg-no-repeat;
  animation: checkmark var(--ice-animation-input, 0.2s) ease-out;
}
.checkbox:indeterminate {
  @apply bg-base-content bg-no-repeat;
  animation: checkmark var(--ice-animation-input, 0.2s) ease-out;
}
.checkbox-primary {
  @apply border-primary [@media(hover:hover)]:hover:border-primary;
}
.checkbox-primary:focus-visible {
  @apply outline-primary;
}
.checkbox-primary:checked, .checkbox-primary[checked=true], .checkbox-primary[aria-checked=true] {
  @apply border-primary bg-primary text-primary-content;
}
.checkbox-info {
  @apply border-info [@media(hover:hover)]:hover:border-info;
}
.checkbox-info:focus-visible {
  @apply outline-info;
}
.checkbox-info:checked, .checkbox-info[checked=true], .checkbox-info[aria-checked=true] {
  @apply border-info bg-info text-info-content;
}
.checkbox-success {
  @apply border-success [@media(hover:hover)]:hover:border-success;
}
.checkbox-success:focus-visible {
  @apply outline-success;
}
.checkbox-success:checked, .checkbox-success[checked=true], .checkbox-success[aria-checked=true] {
  @apply border-success bg-success text-success-content;
}
.checkbox-warning {
  @apply border-warning [@media(hover:hover)]:hover:border-warning;
}
.checkbox-warning:focus-visible {
  @apply outline-warning;
}
.checkbox-warning:checked, .checkbox-warning[checked=true], .checkbox-warning[aria-checked=true] {
  @apply border-warning bg-warning text-warning-content;
}
.checkbox-error {
  @apply border-error [@media(hover:hover)]:hover:border-error;
}
.checkbox-error:focus-visible {
  @apply outline-error;
}
.checkbox-error:checked, .checkbox-error[checked=true], .checkbox-error[aria-checked=true] {
  @apply border-error bg-error text-error-content;
}
.checkbox-neutral {
  @apply border-neutral [@media(hover:hover)]:hover:border-neutral;
}
.checkbox-neutral:focus-visible {
  @apply outline-neutral;
}
.checkbox-neutral:checked, .checkbox-neutral[checked=true], .checkbox-neutral[aria-checked=true] {
  @apply border-neutral bg-neutral text-neutral-content;
}
.checkbox:disabled {
  @apply bg-base-content cursor-not-allowed border-transparent opacity-20;
}

@keyframes checkmark {
  0% {
    background-position-y: 5px;
  }
  50% {
    background-position-y: -2px;
  }
  100% {
    background-position-y: 0;
  }
}
/* backward compatibility */
.checkbox-mark {
  @apply hidden;
}"
`;
